.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,29 @@
+name: Pre-Commit
+
+on:
+  workflow_dispatch:
+  push:
+
+env:
+  GOLANGCI_LINT_VERSION: "v1.50.1"
```

Comment:
I'm thinking if there would be a way to get this golanci-lint version from somewhere, but it looks complicated. 

[Our precommit hook](https://github.com/EnterpriseDB/terraform-provider-biganimal/blob/main/.pre-commit-config.yaml#L45) uses [the pre-commit-golang repo](https://github.com/dnephin/pre-commit-golang) and we do not describe the golangci-lint version there. AFAICS, pre-commit-golang repo also doesn't define it anywhere. So, it's fine as long as we have golangci-lint installed in the path. 

I wish the pre-commit/action took care of installation of the required tools though 🤷 but not possible for now :)

Reply:
@serdardalgic exactly: as described in the [doc](https://github.com/dnephin/pre-commit-golang#available-hooks), we should take care of installing it.

Maybe we could use `latest` as a version, but usually I prefer to pin all the tools involved in our chain (like we already do in the [.pre-commit-config.yaml](https://github.com/EnterpriseDB/terraform-provider-biganimal/blob/main/.pre-commit-config.yaml) file, see the `rev` field)

Reply:
I strongly agree with you @mbiagetti, pinning the version sounds like a sane approach. Thanks for the links!